src/os/bluestore/BlueStore.cc

```diff
@@ -5931,7 +5936,10 @@ void BlueStore::_do_write_small(
 	     << " bstart 0x" << std::hex << bstart << std::dec << dendl;
 
     // can we pad our head/tail out with zeros?
-    uint64_t chunk_size = b->blob.get_chunk_size(b->blob.has_csum(), block_size); //blob csum settings to be applied hence ignoring current config settings for csum enable/disable
+    // blob csum settings to be applied hence ignoring current config
+    // settings for csum enable/disable
+    uint64_t chunk_size = b->dirty_blob().get_chunk_size(
+      b->get_blob().has_csum(), block_size);
```

Comment:
b->get_blob().has_csum() can be replaced with true here as has_csum is called internally